src/metabase/lib/temporal_bucket.cljc

```diff
@@ -227,6 +227,30 @@
   [_query _stage-number _x]
   #{})
 
+(defn- mark-unit [options option-key unit]
+  (cond->> options
+    (some #(= (:unit %) unit) options)
+    (mapv (fn [option]
+            (cond-> option
+              (contains? option option-key) (dissoc option option-key)
```

Comment:
I understand it's not new, but could you still add a comment why the test `(contains? option option-key)` is necessary/beneficial?